### ISSUE TITLE
[BUG] Launch conv kernel with offline ptx failed as invalid argument

### ISSUE THREAD

**Issue by shenzhenghai** [NVIDIA/cutlass#475]
I am trying to runtime load offline compiled ptx using the same CUDA source file and launch kernel using cuLaunchKernel, but examples/16_ampere_tensorop_conv2dfprop failed with driver error code 1. 

```

> CUDA Driver error at .../cutlass/examples/16_ampere_tensorop_conv2dfprop/../../include/cutlass/conv/device/implicit_gemm_convolution.h:285 code=1(invalid argument) "cuLaunchKernel(kernel, grid.x, grid.y, grid.z, block.x, block.y, block.z, smem_size, stream, args, 0)"
> terminate called after throwing an instance of 'std::runtime_error'
>   what():  CUDA driver failed.

```
It's OK to use the same launch call if getting kernel with cudaGetFuncBySymbol.
I also tried examples/00_basic_gemm, both methods run smoothly.

Environment: CUDA 11.6 on Ubuntu 20, RTX 3090
steps to reproduce:
1. Apply the attached patch to latest code
2. change LAUNCH_MODE to 0
3. cd examples/16_ampere_tensorop_conv2dfprop/
4. make
5. cp tmp/*.ptx kernel.ptx
6. change LAUNCH_MODE back to 2
7. make clean && make
8. ./case
It will show the error message as above.

[patch_offline_ptx.log](https://github.com/NVIDIA/cutlass/files/8568354/patch_offline_ptx.log)

**Comment by shenzhenghai** [comment]
Thanks for the quick reply.
I tried `kAnalytic`, got same invalid argument error.

In fact, it can work using the same cuda driver API with the same parameters, just using alternative way (`cudaGetFuncBySymbol`) to retrieve kernel handle (`LAUNCH_MODE=1` in the patch)
It seems executable builtin fatbin might be different than offline compiled one, I also tried following method to get offline ptx/cubin/fatbin, none can work.
1. nvcc compile time saved cubin
2. nvcc compile time saved fatbin
3. cuobjdump ptx from the executable
4. cuobjdump sass from the executable

Probably it need NVIDIA driver or compiler team's help to root cause.

**Comment by shenzhenghai** [comment]
@mnicely May I know any updates for the bug? Thanks.

**Comment by shenzhenghai** [comment]
@mnicely @hwu36 any updates after 2 weeks?
Thanks.

**Comment by hwu36** [comment]
Sorry, I haven't looked into your case again.  I know nvrtc + conv can work since many users make it work.

We are working on enable cutlass conv with python.  You may take a look when we enable it.  See https://github.com/NVIDIA/cutlass/issues/474

**Comment by shenzhenghai** [comment]
@hwu36 I am enabling nvrtc using cutlass conv, encountered the issue. I suspected runtime compiled PTX might not be as expected , so I compared it with nvcc kept PTX, they were almost same. And both PTX got same launching error described in the beginning.
Gemm case were OK using either of the PTX generated by the same flow.

Any example for nvrtc + cutlass conv? waiting for #474?

**Comment by hwu36** [comment]
pycutlass implemented in 2.10 supports to use nvrtc to build conv.
